plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/commands/draw/DrawCoordinateCommand.java

```diff
@@ -0,0 +1,244 @@
+        }
+        useCircle = false;
+    }
+
+    /**
+     * Does the actual job by drawing a 3x3 rectangle or circle at the 
+     * coordinate location
+     * 
+     * @see org.locationtech.udig.project.command.MapCommand#run()
+     */
+    public void run(IProgressMonitor monitor) throws Exception {
+
+        final Point point = getMap().getViewportModel().worldToPixel(coord);
+        Shape shape = useCircle ? new Ellipse2D.Double(point.x, point.y, 3, 3) : 
+            new Rectangle2D.Double(point.x, point.y, 3, 3);
+        
+        if (fill != null) {
+            graphics.setColor(fill);
+            graphics.fill(shape);
+        }
+
+        if (paint != null)
+            graphics.setColor(paint);
+        if (style > -1)
+            graphics.setStroke(style, width);
+        doDraw(shape);
+    }
+
+    /**
+     * actually the Shape is always of type Rectangle2D but we may enhance 
+     * and make it configurable in the future.
+     *  
+     * @param shape
+     */
+    private void doDraw(Shape shape) {
+        if (shape instanceof Rectangle2D) {
+            Rectangle2D rect = (Rectangle2D) shape;
+            graphics.drawRect((int) rect.getX(), (int) rect.getY(), (int) rect.getWidth(),
+                    (int) rect.getHeight());
+            return;
+        }
+
+        if (shape instanceof Ellipse2D) {
+            Ellipse2D ellipse = (Ellipse2D) shape;
+            graphics.drawOval((int) ellipse.getMinX(), (int) ellipse.getMinY(),
+                    (int) ellipse.getWidth(), (int) ellipse.getHeight());
+            return;
+        }
+    }
+
+    /**
+     * @return Returns the paint.
+     */
+    public Paint getPaint() {
+        return paint;
+    }
+
+    /**
+     * @param paint The paint to set.
+     */
+    public void setPaint(Color paint) {
+        this.paint = paint;
+    }
+
+    /**
+     * @return Returns the line style.
+     */
+    public int getLineStyle() {
+        return style;
+    }
+
+    /**
+     * @return Returns the line width.
+     */
+    public int getLineWidth() {
+        return width;
+    }
+
+    /**
+     * Sets the line style
+     * 
+     * @param lineStyle the style of the line
+     * @param lineWidth the width of the line
+     */
+    public void setStroke(int lineStyle, int lineWidth) {
+        this.style = lineStyle;
+        this.width = lineWidth;
+    }
+
+    /**
+     * Sets the color that the shape will be filled with. If fill is null then no fill will be
+     * applied.
+     * 
+     * @param fillColor a color to be used to fill the shapeor null.
+     */
+    public void setFill(Color fillColor) {
+        this.fill = fillColor;
+    }
+
+
+    /**
+     * @return the useCircle
+     */
+    public boolean isUseCircle() {
+        return useCircle;
+    }
+
+    /**
+     * @param useCircle the useCircle to set
+     */
+    public void setUseCircle(boolean useCircle) {
+        this.useCircle = useCircle;
+    }
+
+    @Override
```

Comment:
Yes there is no test case as is the case with all DrawCommands. If you think it absolutely necessary, maybe I can provide a test in the form of a test menu item that draws a circle at a coordinate on a map. But it could be tested manually and not be automated. What do you think?